File: python/packages/ai/teams/ai/moderator/openai_moderator_options.py
"""
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the MIT License.
"""

from dataclasses import dataclass
from typing import Literal, Optional


@dataclass
class OpenAIModeratorOptions:
    """
    Options for the OpenAI based moderator.
    """

    api_key: str
    """
    OpenAI API Key
    """

    moderate: Literal["input", "output", "both"]
    """
    Which parts of the conversation to moderate.
    """

    organization: Optional[str] = None
    """
    Optional. OpenAI organization.
    """

    model: Optional[str] = None
    """
    Optional. OpenAI model to use.
    """

    api_version: Optional[str] = None
    """
    Optional. Azure Content Safety API version.
    """

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/OpenAIModeratorOptions.cs

namespace Microsoft.Teams.AI.AI.Moderator
{
    /// <summary>
    /// Options for the OpenAI Moderator.
    /// </summary>
    public class OpenAIModeratorOptions
    {
        /// <summary>
        /// OpenAI API key.
        /// </summary>
        public string ApiKey { get; set; }

        /// <summary>
        /// Which parts of the conversation to moderate.
        /// </summary>
        public ModerationType Moderate { get; set; }

        /// <summary>
        /// Optional. OpenAI organization.
        /// </summary>
        public string? Organization { get; set; }

        /// <summary>
        /// Optional. OpenAI endpoint.
        /// </summary>
        public string? Endpoint { get; set; }

        /// <summary>
        /// The moderation model to use.
        /// </summary>
        public string? Model { get; set; }

        /// <summary>
        /// Create an instance of the OpenAIModeratorOptions class.
        /// </summary>
        /// <param name="apiKey">OpenAI API key.</param>
        /// <param name="moderate">Which parts of the conversation to moderate.</param>
        public OpenAIModeratorOptions(string apiKey, ModerationType moderate)
        {
            ApiKey = apiKey;
            Moderate = moderate;
        }
    }

    /// <summary>
    /// Which parts of the conversation to moderate.
    /// </summary>
    public enum ModerationType
    {
        /// <summary>
        /// Only moderate input.
        /// </summary>
        Input,

        /// <summary>
        /// Only moderate output.
        /// </summary>
        Output,

        /// <summary>
        /// Moderate both input and output.
        /// </summary>
        Both
    };
}
